src/targets/gpu/fuse_mlir.cpp

```diff
@@ -63,43 +64,47 @@ struct mlir_conv
 };
 MIGRAPHX_REGISTER_OP(mlir_conv);
 
-namespace {
-
-MIGRAPHX_PRED_MATCHER(is_mlir_conv, instruction_ref ins)
+struct mlir_dot
 {
-    if(ins->name() != "convolution")
-        return false;
-    value v    = ins->get_operator().to_value();
-    auto group = v.at("group").to<int>();
-    if(group != 1)
-        return false;
-    // Avoid MLIR assertion: Index < Length && "Invalid index!"
-    if(ins->get_shape().lens().size() != 4)
-        return false;
-    return true;
-}
+    std::string name() const { return "gpu::mlir_dot"; }
+    operation op = make_op("dot");
 
-struct find_conv_pointwise
-{
-    // Find a convolution followed by a pointwise operation.
-    auto matcher() const
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
     {
-        auto convolution =
-            match::skip(match::name("contiguous"))(is_mlir_conv().bind("convolution"));
-        return match::name("pointwise")(match::any_of[match::inputs()](convolution.bind("x")));
+        return pack(f(self.op, "op"));
     }
 
+    shape compute_shape(std::vector<shape> inputs, const std::vector<module_ref>& mods) const
+    {
+        check_shapes{inputs, *this}.packed_or_broadcasted();
+        if(mods.size() != 1)
+            MIGRAPHX_THROW("should have one submodule.");
+        if(inputs.size() < 2)
+            MIGRAPHX_THROW("should have at least two inputs.");
+        auto n = inputs.size();
+        return op.compute_shape({inputs[n - 2], inputs[n - 1]});
+    }
+};
+MIGRAPHX_REGISTER_OP(mlir_dot);
```

Comment:
These could be combined into the same operator:

```cpp
struct mlir_op
{
    operation op;

    template <class Self, class F>
    static auto reflect(Self& self, F f)
    {
        return pack(f(self.op, "op"));
    }

    std::string name() const { return "gpu::mlir_op"; }
    shape compute_shape(std::vector<shape> inputs, const std::vector<module_ref>& mods) const
    {
        check_shapes{inputs, *this}.packed_or_broadcasted();
        if(mods.size() != 1)
            MIGRAPHX_THROW("should have one submodule.");
        if(inputs.size() < 2)
            MIGRAPHX_THROW("should have at least two inputs.");
        auto n = inputs.size();
        return op.compute_shape({inputs[n - 2], inputs[n - 1]});
    }
};
MIGRAPHX_REGISTER_OP(mlir_op);
```

Reply:
Yes could be, if thats preferred.

I'd use mlip_binary_op or something because just mlir_op seems too generic for that specific compute_shape implementation.

Reply:
> I'd use mlir_binary_op or something because just mlir_op seems too generic for that specific compute_shape implementation.

We can make it work with any number of inputs by checking the number of parameters in the submodule, but we can tackle that later when we do different parameters.

Reply:
ok. done.